### ISSUE TITLE
windows 10 20H2

### ISSUE THREAD

**Issue by amidevous** [billziss-gh/winfsp#330]
on windows 10 version 20H2 winfsp causes windows to stop immediately as soon as it is mounted

whether with sshfs or rclone none of the 2 works

I tried with the last 2 stable versions 2020.1 and 2020.2 and the 2 causes this problem

no error code visible and there appears to be no log file in the install folder

thank you for solving this problem

**Comment by amidevous** [comment]
WINFSP INSTALLATION DIRECTORY AND LAUNCHER REGISTRATIONS

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp
    InstallDir    REG_SZ    C:\Program Files (x86)\WinFsp\

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x1

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs.k
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x0

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs.r
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x1


WINFSP DLL REGISTRATIONS

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\Order
    ProviderOrder    REG_SZ    WinFsp.Np,RDPNP,LanmanWorkstation,webclient


HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\WinFsp.Np\NetworkProvider
    Name    REG_SZ    Windows File System Proxy
    ProviderPath    REG_SZ    C:\Program Files (x86)\WinFsp\bin\winfsp-x64.dll
    DeviceName    REG_SZ    \Device\WinFsp.Mup


HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\EventLog\Application\WinFsp
    EventMessageFile    REG_SZ    C:\Program Files (x86)\WinFsp\bin\winfsp-x64.dll
    TypesSupported    REG_DWORD    0x7


WINFSP FSD CONFIGURATION AND STATUS

SERVICE_NAME: WinFsp
        TYPE               : 2  FILE_SYSTEM_DRIVER
        STATE              : 1  STOPPED
        WIN32_EXIT_CODE    : 1077  (0x435)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
[SC] QueryServiceConfig réussite(s)

SERVICE_NAME: WinFsp
        TYPE               : 2  FILE_SYSTEM_DRIVER
        START_TYPE         : 3   DEMAND_START
        ERROR_CONTROL      : 1   NORMAL
        BINARY_PATH_NAME   : \??\C:\Program Files (x86)\WinFsp\bin\winfsp-x64.sys
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : WinFsp
        DEPENDENCIES       :
        SERVICE_START_NAME :

D:(A;;LCRP;;;WD)(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)S:(AU;FA;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;WD)

WINFSP LAUNCHER SERVICE CONFIGURATION AND STATUS

SERVICE_NAME: WinFsp.Launcher
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 4  RUNNING
                                (STOPPABLE, NOT_PAUSABLE, ACCEPTS_SHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
[SC] QueryServiceConfig réussite(s)

SERVICE_NAME: WinFsp.Launcher
        TYPE               : 10  WIN32_OWN_PROCESS
        START_TYPE         : 2   AUTO_START
        ERROR_CONTROL      : 0   IGNORE
        BINARY_PATH_NAME   : "C:\Program Files (x86)\WinFsp\bin\launcher-x64.exe"
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : WinFsp.Launcher
        DEPENDENCIES       :
        SERVICE_START_NAME : LocalSystem

D:(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)S:(AU;FA;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;WD)

OS INFORMATION

Host name: SHADOW-TSFJ6AE4
Operating System Name: Microsoft Windows 10 Home
System version: 10.0.19042 N/A version 19042
Operating System Manufacturer: Microsoft Corporation
Operating system configuration: Stand-alone workstation
Type of operating system version: Multiprocessor Free
Registered owner: Shadow
Registered organization:
Product ID: 00326-10863-40210-AA478
Original installation date: 11/01/2020, 4:41:18 PM
System start time: 01/11/2020, 20:29:21
System Manufacturer: Blade
System model: Shadow Computer
System type: x64-based PC
Processor (s): 1 processor (s) installed.
                                            [01]: Intel64 Family 6 Model 63 Stepping 2 GenuineIntel ~ 2500 MHz
BIOS version: Blade 1.0.0, 01/03/2019
Windows directory: C:\Windows
System directory: C:\Windows\system32
Boot device: \Device\HarddiskVolume2
Regional system option: fr; French (France)
Input locale: fr; French (France)
Time zone: (UTC + 01: 00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna
Total physical memory: 12,287 MB
Available physical memory: 9,929 MB
Virtual memory: maximum size: 14,719 MB
Virtual memory: available: 10,761 MB
Virtual memory: in use: 3,958 MB
Paging file locations: C: \ pagefile.sys
Domain: WORKGROUP
Logon server: \\ SHADOW-TSFJ6AE4
Patch (s): 9 Corrections installed.
                                            [01]: KB4580419
                                            [02]: KB4537759
                                            [03]: KB4557968
                                            [04]: KB4561600
                                            [05]: KB4562830
                                            [06]: KB4566785
                                            [07]: KB4577266
                                            [08]: KB4580325
                                            [09]: KB4579311
Network card (s): 1 network card (s) installed.
                                            [01]: Red Hat VirtIO Ethernet Adapter
                                                  Connection name: Ethernet
                                                  DHCP enabled: Yes
                                                  DHCP server: 10.0.2.1
                                                  IP address (es)
                                                  [01]: 10.0.2.10
                                                  [02]: fe80 :: 4036: cfab: ffd: 94ea
Hyper-V System Requirements: A hypervisor has been detected. The features required for Hyper-V will not be displayed.

**Comment by billziss-gh** [comment]
> on windows 10 version 20H2 winfsp causes windows to stop immediately as soon as it is mounted

Can you please provide more detail:

- Provide a detailed list of steps to cause the problem.
- Provide the expected and observed behaviors.
- What does stop immediately mean?
- If there is a BSOD (system crash) please provide the relevant `memory.dmp` file (usually found in the `\Windows` directory).

**Comment by amidevous** [comment]
as soon as we open the reader up its done

file C:\Windows\memory.dmp file not found

the windows in question and in the cloud

https://shadow.tech

retrun error R:105

https://help.shadow.tech/hc/en-gb/articles/360009699020-R-105-Unfortunately-Windows-encountered-a-crash

**Comment by billziss-gh** [comment]
This looks more like an issue with shadow PC rather than Windows 10 20H2. Perhaps an incompatibility between WinFsp and their system software.

Unfortunately there is little I can do without a memory dump.

**Comment by thecoolreaver** [comment]
please @amidevous if you get any response from shadow in regards to this issue, please say. I am having the same issue which seems to be a change shadow did with there VM system.

**Comment by AdamHasker** [comment]
Sorry for my English. I use Google Translate.

Faced a problem like yours.

I had the DISK-O application installed - does it use and install the WinFsp libraries? They are located in the \Windows\System32\disko folder

I uninstalled the DISK-O app, rebooted my PC and installed WinFsp without issue. Then I reinstalled DISC-O. Everything works without problems.

> on windows 10 version 20H2 winfsp causes windows to stop immediately as soon as it is mounted
> 
> whether with sshfs or rclone none of the 2 works
> 
> I tried with the last 2 stable versions 2020.1 and 2020.2 and the 2 causes this problem
> 
> no error code visible and there appears to be no log file in the install folder
> 
> thank you for solving this problem

Sorry for my English. I use Google Translate.

Faced a problem like yours.

I had the DISK-O application installed - does it use and install the WinFsp libraries? They are located in the \Windows\System32\disko folder

I uninstalled the DISK-O app, rebooted my PC and installed WinFsp without issue. Then I reinstalled DISC-O. Everything works without problems.
